### PR TITLE
PR ResNet

### PR DESCRIPTION
**Pull request by zkdlfrlwl2**
First PR ResNet

### CODE REVIEW

File: ResNet/model/block.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        self.conv3_1_1by1_2 = ConvBlock(dim, dim*4, kernel_size=1)
+        self.downsample = ConvBlock(dim*2, dim*4, kernel_size=1, stride=2)
+
+        self.relu = nn.ReLU()
+        self.bn = nn.BatchNorm2d(num_features=dim*4)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        identity = x
+
+        if self.is_first_layer is True and self.is_first_block is True:
```
Comment (rapidrabbit76) [inline ResNet/model/block.py:113]: 가급적이면 forward path 에서 분기 나누기 말것

File: ResNet/model/model.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+class ResNet(nn.Module):
+
+    def __init__(
+            self,
+            image_channels: int,
+            n_classes: int,
+            model_type: int,
+    ):
+        super().__init__()
+        dim = 64
```
Comment (rapidrabbit76) [inline ResNet/model/model.py]: dim 변수 생성자에서 받아서 사용 하도록 변경 
이런거 하드코딩 하면 나중에 힘듬

**Review by rapidrabbit76 — CHANGES_REQUESTED** [review]
분기는 괜찮은데 
모델 구조 하드코딩은 변경 부탁

File: ResNet/model/model.py (hunk 2 above)
Comment (zkdlfrlwl2) [reply ResNet/model/model.py]: 접수

File: ResNet/model/block.py (hunk 1 above)
Comment (zkdlfrlwl2) [reply ResNet/model/block.py:113]: 모델 구조를 따로 make_layer method로 만들어라고 ? 그래서 그 method 통해서 만들어진 layer를 그대로 forward에서 사용하는 식으로 ?

Comment (zkdlfrlwl2) [reply ResNet/model/block.py:113]: forward 분기를 생성자 쪽으로 올려서 바꾸라는 말인가 ?

Comment (rapidrabbit76) [reply ResNet/model/block.py:113]: 가능하면? 
어쩔수 없으면 분기는 괜찮아
